Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -226,20 +240,45 @@ private void OnErrorsChanged (DataErrorsChangedEventArgs e)
 			ErrorsChanged?.Invoke (this, e);
 		}
 
-		private async void OnVariationsChanged (object sender, EventArgs e)
+		private void OnObjectEditorPropertyChanged (object sender, EditorPropertyChangedEventArgs e)
+		{
+			if (!e.Property.HasVariations ())
+				return;
+
+			OnVariantsChanged (e.Property, EventArgs.Empty);
+		}
+
+		private async void OnVariantsChanged (object sender, EventArgs e)
 		{
+			IPropertyInfo property = sender as IPropertyInfo;
+			if (property == null)
+				property = ((PropertyViewModel)sender).Property;
+
 			using (await AsyncWork.RequestAsyncWork (this)) {
-				PropertyViewModel pvm = (PropertyViewModel) sender;
-				var variations = (await GetVariationsAsync (pvm.Property)).SelectMany (vs => vs).Distinct ();
-				var properties = this.editors
-					.OfType<PropertyViewModel> ()
-					.Where (evm => Equals (evm.Property, pvm.Property) && evm.Variation != null)
-					.ToDictionary (evm => evm.Variation);
+				var variationsTask = GetVariationsAsync (property);
+
+				PropertyViewModel baseVm = null;
+				var properties = new Dictionary<PropertyVariation, PropertyViewModel> ();
+				foreach (PropertyViewModel pvm in this.editors.OfType<PropertyViewModel> ()) {
+					if (!Equals (property, pvm.Property))
```

Comment:
You can move that to a `Where` in the `foreach`.

Reply:
True, but I don't find it that much cleaner for its high cost.